src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/ExplosiveTool.java

```diff
@@ -67,24 +70,35 @@ public ToolUseHandler getItemHandler() {
 
     @ParametersAreNonnullByDefault
     private void breakBlocks(Player p, ItemStack item, Block b, List<Block> blocks, List<ItemStack> drops) {
+        ArrayList<Block> blocksToDestroy = new ArrayList<>();
+
         if (callExplosionEvent.getValue().booleanValue()) {
             BlockExplodeEvent blockExplodeEvent = new BlockExplodeEvent(b, blocks, 0);
             Bukkit.getServer().getPluginManager().callEvent(blockExplodeEvent);
 
             if (!blockExplodeEvent.isCancelled()) {
                 for (Block block : blockExplodeEvent.blockList()) {
                     if (canBreak(p, block)) {
-                        breakBlock(p, item, block, drops);
+                        blocksToDestroy.add(block);
                     }
                 }
             }
         } else {
             for (Block block : blocks) {
                 if (canBreak(p, block)) {
-                    breakBlock(p, item, block, drops);
+                    blocksToDestroy.add(block);
                 }
             }
         }
+
+        ExplosiveToolBreakBlockEvent event = new ExplosiveToolBreakBlockEvent(p, blocksToDestroy, item, this);
+        Bukkit.getServer().getPluginManager().callEvent(event);
+
+        if (!event.isCancelled()){
```

Comment:
space after ending brackets

```suggestion
        if (!event.isCancelled()) {
```